test/jdk/java/net/URL/HandlersPkgPrefix/LargeClasspathWithPkgPrefix.java

```diff
@@ -0,0 +1,137 @@
+                    System.out.println("Application launched with java.protocol.handler.pkgs="
+                                + System.getProperty("java.protocol.handler.pkgs"));
+                    System.out.println("Application launched with classpath: "
+                                + System.getProperty("java.class.path"));
+                    System.out.println("Hello World");
+                }
+            }
+            """;
+
+    public static void main(final String[] args) throws Exception {
+        // dir to which the application main's .class file will be compiled to
+        Path classesDir = Files.createTempDirectory(CWD, "8308184-classes").toAbsolutePath();
+        // dir contains many jars
+        Path libDir = Files.createTempDirectory(CWD, "8308184-libs").toAbsolutePath();
+        Files.createDirectories(libDir);
+
+        // trivial jar file
+        Path jarPath = Path.of(libDir.toString(), "8308184-dummy.jar");
+        createJar(jarPath);
+
+        // create multiple such jar files in the lib dir
+        int numCopies = 1000;
+        for (int i = 1; i <= numCopies; i++) {
+            Path dest = Path.of(libDir.toString(), "8308184-dummy-" + i + ".jar");
+            Files.copy(jarPath, dest);
+        }
+        System.out.println("Created " + numCopies + " jars under " + libDir);
+
+        // create the application's main java file
+        Path fooJavaSrcFile = Path.of(classesDir.toString(), "Foo.java");
+        Files.writeString(fooJavaSrcFile, JAVA_MAIN_CONTENT);
+
+        // compile this java file
+        compile(fooJavaSrcFile, classesDir);
+
+        // Create the classpath string. It is important that the classes directory which contains
+        // application's main class, is at the end of the classpath (or too far into the classpath).
+        // The initial entries in the classpath should be jar files.
+        // constructed classpath is of the form -cp lib/*:classes/
+        // (the * in lib/* is parsed/interpreted by the java launcher and includes all jars in that
+        // directory)
+        String classpath = File.pathSeparator + libDir.toString() + "/*"
+                + File.pathSeparator + classesDir.toString();
+        // launch the application
+        launchApplication(classpath);
+        // test passed successfully, we don't need the lib directory which has too many jars,
+        // anymore. we let the dir stay only if the test fails, for debug purpose
+        libDir.toFile().deleteOnExit();
+    }
+
+    // creates a trivial jar file
+    private static void createJar(Path p) throws Exception {
+        JarBuilder jb = new JarBuilder(p.toString());
+        jb.addEntry("foobar.txt", "foobar".getBytes());
+        jb.build();
+        System.out.println("Created jar at " + p);
+    }
+
+    // javac -d <destDir> <javaFile>
+    private static void compile(Path javaFile, Path destDir) throws Exception {
+        String javacPath = JDKToolFinder.getJDKTool("javac");
```

Comment:
FYI.   `jdk.test.lib.compiler.CompilerUtils` can be used to compile classes in process.

Reply:
Thank you Mandy, this is useful, I wasn't aware of this one. I have now updated the test to use this utility class.